gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/EndpointServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.rest.api.service.v4.impl;
+
+import io.gravitee.plugin.endpoint.EndpointConnectorPlugin;
+import io.gravitee.rest.api.model.platform.plugin.PlatformPluginEntity;
+import io.gravitee.rest.api.service.JsonSchemaService;
+import io.gravitee.rest.api.service.impl.AbstractPluginService;
+import io.gravitee.rest.api.service.v4.EndpointService;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author GraviteeSource Team
+ */
+@Component("EndpointServiceImplV4")
+public class EndpointServiceImpl extends AbstractPluginService<EndpointConnectorPlugin, PlatformPluginEntity> implements EndpointService {
+
+    private final JsonSchemaService jsonSchemaService;
+
+    public EndpointServiceImpl(JsonSchemaService jsonSchemaService) {
+        this.jsonSchemaService = jsonSchemaService;
+    }
+
+    @Override
+    public Set<PlatformPluginEntity> findAll() {
+        return super.list().stream().map(this::convert).collect(Collectors.toSet());
+    }
+
+    @Override
+    public PlatformPluginEntity findById(String endpointPluginId) {
+        EndpointConnectorPlugin resourceDefinition = super.get(endpointPluginId);
+        return convert(resourceDefinition);
+    }
+
+    @Override
+    public String validateEndpointConfiguration(final String endpointPluginId, final String configuration) {
+        PlatformPluginEntity endpointPluginEntity = this.findById(endpointPluginId);
+        return validateConfiguration(endpointPluginEntity, configuration);
+    }
+
+    private String validateConfiguration(PlatformPluginEntity plugin, String configuration) {
+        if (plugin != null && configuration != null) {
+            String schema = getSchema(plugin.getId());
+            return jsonSchemaService.validate(schema, configuration);
+        }
+        return configuration;
+    }
```

Comment:
I'm wondering if this could be moved to `AbstractPluginService` so it can be reused (`EntrypointService`, `PolicyService`, ...)

Reply:
Yes, that makes sense. I did that in an additional commit.